packages/babel-plugin/src/utils/ast.tsx

```diff
@@ -113,29 +113,28 @@ export const getKey = (node: t.Expression) => {
  * @param expression Expression we want to interrogate.
  * @param state Babel state - should house options and meta data used during the transformation.
  */
-export const getInterpolation = <TNode extends {}>(expression: TNode | undefined, state: State) => {
-  if (!state.declarations) {
-    return expression;
-  }
-
+export const getInterpolation = <TNode extends {}>(
+  expression: TNode | undefined,
+  meta: Metadata
+) => {
   let value: t.Node | undefined | null = undefined;
 
   if (t.isIdentifier(expression)) {
-    const binding = state.declarations[expression.name];
-
-    if (t.isVariableDeclaration(binding) && binding.kind === 'const') {
-      value = binding.declarations[0].init;
+    const binding = meta.parentPath.scope.getBinding(expression.name);
+    if (binding && binding.kind === 'const' && t.isVariableDeclarator(binding.path.node)) {
+      value = binding.path.node.init;
     }
   } else if (t.isMemberExpression(expression)) {
     const { accessPath, bindingIdentifier } = getMemberExpressionMeta(expression);
-    const binding = state.declarations[bindingIdentifier.name];
+    const binding = meta.parentPath.scope.getBinding(bindingIdentifier.name);
 
     if (
-      t.isVariableDeclaration(binding) &&
+      binding &&
       binding.kind === 'const' &&
-      t.isObjectExpression(binding.declarations[0].init)
+      t.isVariableDeclarator(binding.path.node) &&
```

Comment:
nit: should we extract this into a function? 🤔 We have used it twice atleast

```typescript
const isBindedVariableDeclarator = (binding, kind = 'const') => binding && binding.kind === kind && t.isVariableDeclarator(binding.path.node);
```

and probably for this too `binding.path.node.init`

Reply:
good idea